File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/common/jwt/entity/RefreshToken.java
package com.multicampus.gamesungcoding.a11ymarketserver.common.jwt.entity;

import com.multicampus.gamesungcoding.a11ymarketserver.common.id.UuidV7;
import com.multicampus.gamesungcoding.a11ymarketserver.feature.user.entity.Users;
import jakarta.persistence.*;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Getter
@NoArgsConstructor(access = lombok.AccessLevel.PROTECTED)
public class RefreshToken {
    @Id
    @UuidV7
    @Column(length = 16, updatable = false, nullable = false)
    private UUID refreshTokenId;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "user_id", nullable = false, unique = true)
    @OnDelete(action = OnDeleteAction.CASCADE)
    private Users user;

    @Column(nullable = false, unique = true)
    private String token;

    @Column(nullable = false)
    private LocalDateTime expiryDate;

    @Builder
    private RefreshToken(Users user, String token, LocalDateTime expiryDate) {
        this.user = user;
        this.token = token;
        this.expiryDate = expiryDate;
    }

    public void updateToken(String token, LocalDateTime expiryDate) {
        this.token = token;
        this.expiryDate = expiryDate;
    }
}

File: src/main/kotlin/com/multicampus/gamesungcoding/a11ymarketserver/common/jwt/entity/RefreshToken.kt
package com.multicampus.gamesungcoding.a11ymarketserver.common.jwt.entity

import com.multicampus.gamesungcoding.a11ymarketserver.common.id.UuidV7
import com.multicampus.gamesungcoding.a11ymarketserver.feature.user.entity.Users
import jakarta.persistence.*
import org.hibernate.annotations.OnDelete
import org.hibernate.annotations.OnDeleteAction
import java.time.LocalDateTime
import java.util.*

@Entity
class RefreshToken(
    @OneToOne(cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false, unique = true)
    @OnDelete(action = OnDeleteAction.CASCADE)
    var user: Users,

    @Column(nullable = false, unique = true)
    var token: String,

    @Column(nullable = false)
    var expiryDate: LocalDateTime
) {
    @Id
    @UuidV7
    @Column(length = 16, updatable = false, nullable = false)
    var refreshTokenId: UUID? = null
        private set

    fun updateToken(token: String, expiryDate: LocalDateTime) {
        this.token = token
        this.expiryDate = expiryDate
    }
}
